### PR TITLE
Create test-violations.py

### PR DESCRIPTION
**Pull request by artem-fedorov**
Thank you for submitting a pull request to the WebGoat!

### CODE REVIEW

File: test-violations.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+from attrs import define
+from attrs import field
+
+
+@attr.s(auto_attribs=True)
+class ExampleClass(object):
+    # ruleid:attr-mutable-initializer
+    empty_dict = {}
```
Comment (May-Reuven) [inline test-violations.py:8]: :exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator'.
Severity: Medium

File: test-violations.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    # ruleid:attr-mutable-initializer
+    empty_list = []
```
Comment (May-Reuven) [inline test-violations.py:10]: :exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator'.
Severity: Medium

File: test-violations.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    # ruleid:attr-mutable-initializer
+    somedict = dict()
```
Comment (May-Reuven) [inline test-violations.py:12]: :exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator'.
Severity: Medium

File: test-violations.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    # ruleid:attr-mutable-initializer
+    somelist = list()
```
Comment (May-Reuven) [inline test-violations.py:14]: :exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator'.
Severity: Medium

File: test-violations.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    # ruleid:attr-mutable-initializer
+    someset = set()
```
Comment (May-Reuven) [inline test-violations.py:16]: :exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator'.
Severity: Medium

File: test-violations.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    # ok:attr-mutable-initializer
+    ex_good1 = attr.ib(factory=dict)
+    # ok:attr-mutable-initializer
+    ex_good2 = attr.ib(factory=list)
+    # ok:attr-mutable-initializer
+    ex_good3 = attr.ib(factory=set)
+    # ruleid:attr-mutable-initializer
+    myset = {1, 2, 3}
```
Comment (May-Reuven) [inline test-violations.py:24]: :exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator'.
Severity: Medium

File: test-violations.py (hunk 1 above)
Comment (OmerCalfon) [inline test-violations.py:8]: :exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator. Multiple instances of this class will re-use the same data structure, which is likely not the desired behavior. Consider instead: replace assignment to mutable initializer (ex. dict() or {}) with attr.ib(factory=type) where type is dict, set, or list'.
 
Severity: Medium

File: test-violations.py (hunk 2 above)
Comment (OmerCalfon) [inline test-violations.py:10]: :exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator. Multiple instances of this class will re-use the same data structure, which is likely not the desired behavior. Consider instead: replace assignment to mutable initializer (ex. dict() or {}) with attr.ib(factory=type) where type is dict, set, or list'.
 
Severity: Medium

File: test-violations.py (hunk 3 above)
Comment (OmerCalfon) [inline test-violations.py:12]: :exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator. Multiple instances of this class will re-use the same data structure, which is likely not the desired behavior. Consider instead: replace assignment to mutable initializer (ex. dict() or {}) with attr.ib(factory=type) where type is dict, set, or list'.
 
Severity: Medium

File: test-violations.py (hunk 4 above)
Comment (OmerCalfon) [inline test-violations.py:14]: :exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator. Multiple instances of this class will re-use the same data structure, which is likely not the desired behavior. Consider instead: replace assignment to mutable initializer (ex. dict() or {}) with attr.ib(factory=type) where type is dict, set, or list'.
 
Severity: Medium

File: test-violations.py (hunk 5 above)
Comment (OmerCalfon) [inline test-violations.py:16]: :exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator. Multiple instances of this class will re-use the same data structure, which is likely not the desired behavior. Consider instead: replace assignment to mutable initializer (ex. dict() or {}) with attr.ib(factory=type) where type is dict, set, or list'.
 
Severity: Medium

File: test-violations.py (hunk 6 above)
Comment (OmerCalfon) [inline test-violations.py:24]: :exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator. Multiple instances of this class will re-use the same data structure, which is likely not the desired behavior. Consider instead: replace assignment to mutable initializer (ex. dict() or {}) with attr.ib(factory=type) where type is dict, set, or list'.
 
Severity: Medium

File: test-violations.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+    def bar(self) -> int:
+        # ok:attr-mutable-initializer
+        thisset = set()
+
+
+@define
+class ExampleClass(object):
+    # ruleid:attr-mutable-initializer
+    empty_dict = {}
```
Comment (OmerCalfon) [inline test-violations.py:38]: :exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator. Multiple instances of this class will re-use the same data structure, which is likely not the desired behavior. Consider instead: replace assignment to mutable initializer (ex. dict() or {}) with attr.ib(factory=type) where type is dict, set, or list'.
 
Severity: Medium

File: test-violations.py [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    # ruleid:attr-mutable-initializer
+    empty_list = []
```
Comment (OmerCalfon) [inline test-violations.py:40]: :exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator. Multiple instances of this class will re-use the same data structure, which is likely not the desired behavior. Consider instead: replace assignment to mutable initializer (ex. dict() or {}) with attr.ib(factory=type) where type is dict, set, or list'.
 
Severity: Medium

File: test-violations.py [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    # ruleid:attr-mutable-initializer
+    somedict = dict()
```
Comment (OmerCalfon) [inline test-violations.py:42]: :exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator. Multiple instances of this class will re-use the same data structure, which is likely not the desired behavior. Consider instead: replace assignment to mutable initializer (ex. dict() or {}) with attr.ib(factory=type) where type is dict, set, or list'.
 
Severity: Medium

File: test-violations.py [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+    # ruleid:attr-mutable-initializer
+    somelist = list()
```
Comment (OmerCalfon) [inline test-violations.py:44]: :exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator. Multiple instances of this class will re-use the same data structure, which is likely not the desired behavior. Consider instead: replace assignment to mutable initializer (ex. dict() or {}) with attr.ib(factory=type) where type is dict, set, or list'.
 
Severity: Medium

File: test-violations.py [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+    # ruleid:attr-mutable-initializer
+    someset = set()
```
Comment (OmerCalfon) [inline test-violations.py:46]: :exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator. Multiple instances of this class will re-use the same data structure, which is likely not the desired behavior. Consider instead: replace assignment to mutable initializer (ex. dict() or {}) with attr.ib(factory=type) where type is dict, set, or list'.
 
Severity: Medium

File: test-violations.py [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+    # ok:attr-mutable-initializer
+    ex_good1 = field(factory=dict)
+    # ok:attr-mutable-initializer
+    ex_good2 = field(factory=list)
+    # ok:attr-mutable-initializer
+    ex_good3 = field(factory=set)
+    # ruleid:attr-mutable-initializer
+    myset = {1, 2, 3}
```
Comment (OmerCalfon) [inline test-violations.py:54]: :exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator. Multiple instances of this class will re-use the same data structure, which is likely not the desired behavior. Consider instead: replace assignment to mutable initializer (ex. dict() or {}) with attr.ib(factory=type) where type is dict, set, or list'.
 
Severity: Medium

**Pull request closed without merging**